src/components/DocumentationTopic/DocumentationNav/VersionToggle.vue

```diff
@@ -0,0 +1,283 @@
+    <div :class="{ 'version-toggle-container': manyVersions }">
+      <!-- Faux element to get width of select, with current element-->
+      <select
+        class="version-dropdown version-sizer"
+        ref="version-sizer"
+        aria-hidden="true"
+        tabindex="-1"
+      >
+        <option selected>{{versionModel}}</option>
+      </select>
+      <!-- Faux element is above -->
+      <label
+        :for="versionList ? 'version-toggle' : null"
+        class="nav-menu-setting-label"
+      >Version:</label>
+      <select
+        v-if="versionList.length >1 "
+        id="version-toggle"
+        class="version-dropdown nav-menu-link"
+        :style="`width: ${adjustedWidth}px`"
+        v-model="versionModel"
+        @change="pushRoute(versionRoute)"
+      >
+         <option v-for="version in versionList"
+        v-bind:key="version"
+        :value="version">
+        {{version}}
+        </option>
+      </select>
+
+      <span
+        v-else-if="singleVersionPage"
+        class="nav-menu-toggle-none current-version"
+        aria-current="page"
+      >{{singleVersionPage}}</span>
+      <InlineChevronDownIcon v-if="manyVersions" class="toggle-icon icon-inline" />
+    </div>
+
+  </NavMenuItemBase>
+</template>
+
+<script>
+import DocumentationTopicStore from 'docc-render/stores/DocumentationTopicStore';
+import { waitFrames } from 'docc-render/utils/loading';
+import debounce from 'docc-render/utils/debounce';
+import NavMenuItemBase from 'docc-render/components/NavMenuItemBase.vue';
+import InlineChevronDownIcon from 'theme/components/Icons/InlineChevronDownIcon.vue';
+
+export default {
+  name: 'VersionToggle',
+  components: { InlineChevronDownIcon, NavMenuItemBase },
+  inject: {
+    store: {
+      default() {
+        return {
+          setPreferredVersion() {},
+        };
+      },
+    },
+  },
+  props: {
+    versionList: {
+      type: Array,
+      required: false,
+    },
+  },
+  data() {
+    return {
+      adjustedWidth: 0,
+      versionModel: null,
+    };
+  },
+  mounted() {
+    // on resize, re-calculate the width of the select.
+    const cb = debounce(async () => {
+      // we wait for 3 frames, as that is the minimum it takes
+      // for the browser orientation-change transitions to finish
+      await waitFrames(3);
+      this.calculateSelectWidth();
+    }, 150, true);
+    window.addEventListener('resize', cb);
+    window.addEventListener('orientationchange', cb);
+    this.$once('hook:beforeDestroy', () => {
+      window.removeEventListener('resize', cb);
+      window.removeEventListener('orientationchange', cb);
+    });
+  },
+  updated() {
+    this.versionModel = this.currentVersion;
+  },
+  watch: {
+    // interfaceversion: {
+    //   immediate: true,
+    //   handler(version) {
+    //     this.versionModel = version;
+    //   },
+    // },
+    versionModel: {
+      immediate: true,
+      handler: 'calculateSelectWidth',
+    },
+  },
+  methods: {
+    /**
```

Comment:
I see that you reused the methods from the `LanguageToggle` component. Do you think it makes sense for us to abstract these out into a helper file so we don't repeat the same code for similar components?

Reply:
We should in general refactor those components to a more generic one, but its probably out of the scope of this.